### PR TITLE
Update pySigma

### PR DESCRIPTION
**Pull request by mostafa**
This PR updates the pySigma version along with the package version. I also fixed the breaking change introduced in [pySigma v0.9.10](https://github.com/SigmaHQ/pySigma/releases/tag/v0.9.10) to the processing pipeline class variables: `last_processing_pipeline`. I also bumped the version to `v0.10.0`, so that we can release a new version of this backend after merging the current PR.

### CODE REVIEW

**Review by romain-gaillard — APPROVED** [review]
LGTM! (Beside the tests failing though 😵‍💫)

**Comment by mostafa** [comment]
@kelnage
I am aware of the issue of breaking changes in the pySigma project and am trying to fix it.

File: pyproject.toml [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-packages = [
-    { include = "sigma" }
-]
+packages = [{ include = "sigma" }]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-pysigma = "^0.9.11"
+pysigma = "^0.10.4"
 pysigma-pipeline-sysmon = "^1.0.1"
```
Comment (kelnage) [inline pyproject.toml:17]: Given pysigma-pipeline-sysmon 1.0.1 still depends on pySigma 0.9, I would suggest updating it to 1.0.3 also.

**Review by romain-gaillard — APPROVED** [review]
LGTM! 😄

**Review by kelnage — COMMENTED** [review]
Just a minor comment re: updating other relevant dependencies. You could also usefully update other dependencies in this PR too (mypy and coverage).

File: pyproject.toml (hunk 1 above)
Comment (mostafa) [reply pyproject.toml:17]: It's already up to date in the poetry lock file, but I'll change this file as well.

Comment (mostafa) [reply pyproject.toml]: Done! ✅

**Review by kelnage — APPROVED** [review]
LGTM! Nice fix :+1:

**Pull request merged**